src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1429,11 +1426,7 @@ pub fn parse_dep_info(
             }
         })
         .collect::<Result<Vec<_>, _>>()?;
-    if paths.is_empty() {
-        Ok(None)
-    } else {
-        Ok(Some(paths))
-    }
+    Ok(Some(paths))
```

Comment:
Could you comment on how come this change was necessary? Is it because package files were filtered out for registry deps?

I'm trying to remember why this case was here but I'm coming up with a blank...

Reply:
It looks like the `paths.is_empty()` logic was added in https://github.com/rust-lang/cargo/pull/4788 with no comment as to why. The previous logic looks like what's added in this PR. So long as all tests pass seems reasonable to me!

Reply:
Yea, I couldn't figure out why this check was added in 4788.  The main gist is that when building a registry dependency with current rustc, the list would end up empty (everything gets filtered out), and cargo would treat it as "dirty" when it should be fine.  By instead returning an empty list, it treats it correctly.